src/ui/reducers/possibleBreakpoints.ts

```diff
@@ -125,37 +127,20 @@ export const getPossibleBreakpointsForSelectedSource = (state: UIState): Locatio
   return getPossibleBreakpointsForSource(state, sourceId) || [];
 };
 
-export const getBreakableLinesForSource = createSelector(
-  (state: UIState, sourceId: string) =>
-    state.possibleBreakpoints.entities[sourceId]?.possibleBreakpoints,
-  possibleBreakpoints => {
-    if (!possibleBreakpoints) {
-      return null;
-    }
-    return computeBreakableLines(possibleBreakpoints);
-  },
-  {
-    memoizeOptions: {
-      maxSize: 20,
-    },
-  }
-);
+export const getBreakableLinesForSource = (state: UIState, sourceId: string) => {
+  return adapterSelectors.selectById(state, sourceId)?.breakableLines;
+};
 
-const computeBreakableLines = (possibleBreakpoints: Location[]) => {
+export const computeBreakableLines = (possibleBreakpoints: Location[]) => {
   return uniq(possibleBreakpoints?.map(location => location.line));
 };
 
 export const getBreakableLinesForSelectedSource = (state: UIState) => {
   const sourceId = state.sources.selectedLocation?.sourceId;
   if (!sourceId) {
-    return null;
-  }
-  const possibleBreakpoints = adapterSelectors.selectById(state, sourceId)?.possibleBreakpoints;
-  if (!possibleBreakpoints) {
-    return null;
+    return [];
```

Comment:
:hand: Move this out so it's a constant declared _outside_ the selector, like `const NO_BREAKPOINTS = []`, and then do `return NO_BREAKPOINTS` inside - we actually saw this same problem causing extra re-renders yesterday.